app/controllers/tickets_controller.rb

```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+
+class TicketsController < ApplicationController
+  def index
+    tickets = Ticket.all
+    render json: tickets
+  end
+
+  def show
+    ticket = Ticket.find(params[:id])
+    render json: ticket
+  end
+
+  def create
+    if seat_for_screening_is_taken?
+      render json: { error: 'Ticket taken' }
+    else
+      ticket = Ticket.new(ticket_params)
+      if ticket.save
+        render json: ticket, status: :created
+      else
+        render json: ticket.errors, status: :bad_request
+      end
+    end
+  end
+
+  # I did this so that the seat_for_screening_is_taken doesnt throw error when you try to edit ticket type or price with
+  # existing screening and seat id
+  # If there is a clearer way to write this feel free to correct me
+  def update
+    ticket = Ticket.find(params[:id])
+    if seat_for_screening_is_taken?
+      if ticket.screening_id == params[:screening_id].to_i && ticket.seat_id == params[:seat_id].to_i
```

Comment:
What exactly do this check? If you already have the same `screening_id` and `seat_id` in the ticket as in the params then it will trigger no update (no change), right?

Reply:
this should first be checking if the :screening_id and :seat_id are already in the Tickets db. If so, I added this to check if we changed those params. If not, and we only changed type or price then I let it pass. If the ticket_params[:screening_id] are not the same as the screening_id of the ticket we searched for at line 31 then it means we are trying to edit params to a seat at screening that is already taken.

Reply:
When I tried to do this the same way as in create method whenever I tried to change only type or price leaving the same screening_id and seat_id it would throw the seat already taken error

Reply:
Oh and there is a mistake, it should be ticket.screening_id == ticket_params[:screening_id]